aries_vcx/src/common/proofs/proof_request.rs

```diff
@@ -36,16 +36,27 @@ impl ProofRequestData {
         })
     }
 
-    pub fn set_requested_attributes_as_string(mut self, requested_attrs: String) -> VcxResult<Self> {
-        match serde_json::from_str::<HashMap<String, AttrInfo>>(&requested_attrs) {
+    // Support requested_attributes in 2 formats
+    // Referent-less format, deprecated:
+    // [ { name: 'age' }, { name: 'balance' } ]
+    // Explicit referent format:
+    // { 'attr_age': { 'name': 'age' }, 'attr_balance': { 'name': 'balance' } }
+    // todo: move String parsing to libvcx_core
+    pub fn set_requested_attributes_as_string(mut self, requested_attributes: String) -> VcxResult<Self> {
+        match serde_json::from_str::<HashMap<String, AttrInfo>>(&requested_attributes) {
             Ok(attrs) => self.requested_attributes = attrs,
             Err(_err) => {
-                let requested_attributes: Vec<AttrInfo> = ::serde_json::from_str(&requested_attrs).map_err(|err| {
-                    AriesVcxError::from_msg(
-                        AriesVcxErrorKind::InvalidJson,
-                        format!("Invalid Requested Attributes: {:?}, err: {:?}", requested_attrs, err),
-                    )
-                })?;
+                warn!("Requested attributes are not in referent format. Trying to parse as array of attributes (deprecated).");
+                let requested_attributes: Vec<AttrInfo> =
+                    ::serde_json::from_str(&requested_attributes).map_err(|err| {
+                        AriesVcxError::from_msg(
+                            AriesVcxErrorKind::InvalidJson,
+                            format!(
+                                "Invalid Requested Attributes: {:?}, err: {:?}",
+                                requested_attributes, err
+                            ),
+                        )
+                    })?;
```

Comment:
Maybe it'd be more idiomatic to create an enum with untagged deserialization trying the `HashMap` first and then `Vec`.

Reply:
The vector format is legacy and deprecated, will go away. I don't think it's worthy improving readability in this case